backend/auth-service/src/main/kotlin/com/nosql/authservice/component/impl/DefaultUserComponent.kt

```diff
@@ -0,0 +1,109 @@
+package com.nosql.authservice.component.impl
+
+import com.nosql.authservice.common.exception.DataIntegrityViolationApplicationException
+import com.nosql.authservice.common.exception.enumerator.ApplicationExceptionCodeEnum
+import com.nosql.authservice.common.logger.logFailed
+import com.nosql.authservice.common.logger.logSuccess
+import com.nosql.authservice.component.UserComponent
+import com.nosql.authservice.entity.UserEntity
+import com.nosql.authservice.common.exception.ApplicationException
+import com.nosql.authservice.common.exception.NotFoundException
+import com.nosql.authservice.common.logger.logBefore
+import com.nosql.authservice.repository.UserRepository
+import kotlinx.coroutines.reactor.awaitSingle
+import com.nosql.authservice.common.logger.logger
+import com.nosql.authservice.dto.UserDto
+import com.nosql.authservice.exception.UserAlreadyExistsException
+import org.slf4j.Logger
+import org.springframework.dao.DataIntegrityViolationException
+import org.springframework.stereotype.Component
+import org.springframework.transaction.annotation.Transactional
+import reactor.core.publisher.Mono
+
+@Component
+@Transactional
+class DefaultUserComponent(
+    private val userRepository: UserRepository,
+) : UserComponent {
+
+    private val logger: Logger by logger()
```

Comment:
I think you could simplify it:
```suggestion
    private val log: Logger by logger()
```

Reply:
fixed